### DISCUSSION TITLE
Support LDAPS ( TLS )  in System.DirectoryServices.Protocols ON Linux

### DISCUSSION THREAD

**Discussion by carlosromanbarrado** [dotnet/runtime#103242] [Q&A]
Hello, 

   I have spent several days trying to connect using **LdapConnection** in Linux VMs again LDAP Servers protected by LDAPS. 
   I could not, but i can on Windows.

   I can connect LDAP on boths, Linux & Windows. 
 
   I´m not sure if i´m doing something wrong or the reason is that LDAPs is not supported on Linux on NET 8.
 
   I´m allways get an Exception:   "L**DAP Server is unavailable**"
 
My Code is very simple:
    
NET 8.0
System.DirectoryServices.Protocols 8.0.0

---
    string ldapServer = "MYSERVER";  // Real Data in my tests
    string ldapPort = "636";
    string userConn = "userDN"; // Real Data in my tests
    string password = "passwords"; // Real Data in my tests
    

    ldapsettings = new LdapDirectoryIdentifier(ldapServer,Int32.Parse(ldapPort),false,false);
    ldapConn = new LdapConnection(ldapsettings);
    ldapConn.Credential = new NetworkCredential(userConn, password);
    ldapConn.AuthType = AuthType.Basic;
    LdapSessionOptions options = ldapConn.SessionOptions;
    options.ProtocolVersion = 3;

    ldapConn.Bind();
---

  Let me for any question.

Many Thanks,
Carlos Román

**Comment by teo-tsirpanis** [comment]
@dotnet/area-system-directoryservices

**Comment by buyaa-n** [comment]
Not sure for sure, you might be using a LdapOption that is not supported on Linux, check: https://github.com/dotnet/runtime/blob/d5d244823211f2a9603118b7152fa6f3f36636f7/src/libraries/Common/src/Interop/Interop.Ldap.cs#L134-L149
For example you might be using `LdapSessionOptions.VerifyServerCertificate` callback that [is not supported on Linus](https://github.com/dotnet/runtime/issues/60972)
